src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -75,27 +76,19 @@ private async Task<string> GetRegionAsync(RequestContext requestContext)
                 {
                     { "Metadata", "true" }
                 };
-                
-                HttpResponse response = await _httpManager.SendGetAsync(BuildImdsUri(DefaultApiVersion), headers, logger).ConfigureAwait(false);
+
+                HttpResponse response = await _httpManager.SendGetWithTimeoutAsync(BuildImdsUri(DefaultApiVersion), headers, logger, TimeoutInMs).ConfigureAwait(false);
```

Comment:
I don't understand what's happening here. I see that you've eliminated the retry and that's ok, but I don't get the whole timeout. If the 2 seconds run out, this method will throw. Is this expected? 

Also, the timeout logic is wrong. Every AcquireToken* methdo, including `AcquireTokenForClient` have the option of using a CancellationToken, which the user provides. See `ExecuteAsync(CancellationToken)`. 

So if the user sets a CancellationToken of 1 second, but your internal logic has a timeout of 2 seconds, you need to obey the user's timeout. This sounds complicated but luckly .NET has mechanism for this, see 

```csharp
 CancellationTokenSource.CreateLinkedTokenSource(
                    internalCancellationToken, 
                    externalCancellationToken);
```

Reply:
Discussed offline and updated